[Zhu Minghui] iP

ZMinghuiZ opened the pull request.

ZMinghuiZ committed:
Level 0: Rename to Zuke, add greet and exit

ZMinghuiZ committed:
Add echo feature

ZMinghuiZ committed:
Add store task and list task feature

ZMinghuiZ committed:
Add mark and unmark feature

ZMinghuiZ committed:
Add support for tracking Todo, Deadline and Event

ZMinghuiZ committed:
Resolve cyclic dependency

Resolved cyclic dependency between TaskList class and ResponseManager class by make only ResponseManager depend on TaskList

ZMinghuiZ committed:
Add automated text-ui-test

ZMinghuiZ committed:
Improve code quality, resolve cyclic dependency

make the code more readable and solve the further cyclic dependency between ResponseManager, TaskList, and ZukeLogic that induced from previous resolve of cyclic dependency

Vavinan commented on src/main/java/ResponseManager.java:
```diff
@@ -0,0 +1,83 @@
+            "Got it. I've added this task:\n";
+    private static final String MARK_ERROR_MESSAGE =
+            "please enter a valid index for mark!\n";
+    private static final String UNMARK_ERROR_MESSAGE =
```
I like how you used capital case for constant variables

Vavinan commented on src/main/java/MessageDecoder.java:
```diff
@@ -0,0 +1,47 @@
+    }
+    public static String[] decodeDeadline(String message) {
+        int indexOfDeadlinePrefix = message.indexOf(DEADLINE_PREFIX);
+        String[] decoded = new String[DEADLINE_INFO_COUNT];
```
Perhaps a more intuitive variable name here ,like decodedInput would be better.

Vavinan commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,54 @@
+        this.tasks = new Task[100];
+    }
+
+    public void add(Task task) {
```
Perhaps a more intuitive function name here like addTask.

Vavinan commented on src/main/java/ResponseManager.java:
```diff
@@ -0,0 +1,83 @@
+        System.out.println(INDENTATION_LINE);
+        System.out.println(response + INDENTATION_LINE);
+    }
+    public static void greet() {
```
I like how you made separate class to manage print statements and functions to handle the tasks.

Vavinan reviewed (commented):
Overall the Coding standard is good, minor issue is some variable names and function names. Good job!

ZMinghuiZ commented:
> Overall the Coding standard is good, minor issue is some variable names and function names. Good job!

Thanks for your advice! Will take note of the issues.

ZMinghuiZ commented:
> Overall the Coding standard is good, minor issue is some variable names and function names. Good job!

Thanks for your advice! Will take note of the issues.

Fureimi commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,54 @@
+public class TaskList {
+    private int numberOfTask;
```
Maybe a more intuitive name would be idxOfTask or indexOfTask?

Fureimi commented on src/main/java/Zuke.java:
```diff
@@ -0,0 +1,6 @@
+public class Zuke {
+    public static void main(String[] args) {
+        ResponseManager.greet();
+        new ZukeLogic().chattingStart();
```
I like how the logic for the bot is encapsulated in chattingStart()

Fureimi commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+public class Deadline extends Task {
+    protected String dueTime;
+    private static final int TASK_NAME_INDEX = 0;
+    private static final int DUE_DATE_INDEX = 1;
```
I like how you made the constants into private static final ints and also named them in all caps

Fureimi commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+        this.dueTime = dueTime;
+    }
+
+    @Override
```
I like how you used an override here, helps enhance readability. Also noticed this in other parts of your code.

Fureimi reviewed (commented):
Overall good and clean code, not much to comment on, keep it up!

ZMinghuiZ commented:
> Overall good and clean code, not much to comment on, keep it up!

Thank you!

ZMinghuiZ committed:
Add exception to handle errors

ZMinghuiZ committed:
Merge branch 'branch-Level-5'

ZMinghuiZ committed:
Manage file with packages

ZMinghuiZ committed:
Merge branch 'branch-A-Packages'

ZMinghuiZ committed:
Debug decode event task method; Format the code

aditichadha1310 commented on src/main/java/ZukeLogic.java:
```diff
@@ -0,0 +1,99 @@
+        ResponseManager.sayGoodbye();
+    }
+
+    public void chattingStart() {
```
Avoid long methods. Take corrective action when method exceeds 30 lines of code.

ZMinghuiZ replied:
Thank you for the advice! Will refactor the method to make it more readable

aditichadha1310 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+
+public class Deadline extends Task {
+    protected String dueTime;
+    private static final int TASK_NAME_INDEX = 0;
+    private static final int DUE_DATE_INDEX = 1;
```
Good job on the use of access modifiers.

ZMinghuiZ replied:
Thank you

aditichadha1310 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        String status = isDone ? "X" : " ";
```
Avoid usage of magic strings as much as possible

ZMinghuiZ replied:
Thank you, will refactor the magic strings!

aditichadha1310 reviewed (commented):
Overall, Good job!

ZMinghuiZ committed:
Add delete task feature

ZMinghuiZ committed:
Simplify code

ZMinghuiZ committed:
Add save and load data feature

ZMinghuiZ committed:
Merge branch 'branch-Level-6'

ZMinghuiZ committed:
Resolve merge conflict and merge level-7

ZMinghuiZ committed:
Remove data folder; Update gitignore

Remove unecessary folder from remote repo and update .gitignore to prevent uploading the folder in the future

ZMinghuiZ committed:
Add store data to text feature; Build Jar file

ZMinghuiZ committed:
Refactor program to make it more Object-oriented

ZMinghuiZ committed:
Allow Zuke to recognise date and time input

ZMinghuiZ committed:
Add find keyword feature

find tasks with descriptions that contains the keyword

ZMinghuiZ committed:
Add JavaDoc to most non-private methods

ZMinghuiZ committed:
Merge pull request #1 from ZMinghuiZ/branch-Level-8

Allow Zuke to recognise date and time input

ZMinghuiZ committed:
Merge branch 'master' into branch-Level-9

ZMinghuiZ committed:
Merge branch 'master' into branch-A-JavaDoc

ZMinghuiZ committed:
Merge pull request #2 from ZMinghuiZ/branch-Level-9

Add find keyword feature

ZMinghuiZ committed:
Merge branch 'master' into branch-A-JavaDoc

ZMinghuiZ committed:
Add more JavaDoc

ZMinghuiZ committed:
Merge pull request #3 from ZMinghuiZ/branch-A-JavaDoc

Add JavaDoc to most non-private methods

ZMinghuiZ committed:
Adjust the JavaDocs to follow the course standard

ZMinghuiZ committed:
Adjust DataManager to output human editable file

ZMinghuiZ committed:
Add find task with specific time feature

ZMinghuiZ committed:
Add help command; Customize Zuke's reponse

ZMinghuiZ committed:
Enhance time identification; Refine JavaDoc

Enhance Zuke's ability to recognize date and time

ZMinghuiZ committed:
Update README with user guide

ZMinghuiZ committed:
Enhance user guide format

ZMinghuiZ committed:
Restructured the packages

ZMinghuiZ committed:
Debug date and time recognization of Zuke

ZMinghuiZ committed:
JavaDoc the classes

ZMinghuiZ committed:
Customise Zuke's response message

ZMinghuiZ committed:
Enhance clarity of user guide

ZMinghuiZ committed:
Format user guide

ZMinghuiZ committed:
Resolve bugs in the program

ZMinghuiZ committed:
Update link in the user guide

ZMinghuiZ committed:
Update user guide with customized Zuke response

ZMinghuiZ committed:
Add more explaination to user guide

ZMinghuiZ committed:
Update text-ui-test

ZMinghuiZ committed:
Update date time input instruction in user guide
